### PR TITLE
Run mypy on some more files

### PR DESCRIPTION
**Pull request by MichaelSnowden**
-  [x] `make format && make documentation` has been run.

## What's changed

This PR contains a batch of fixes to mypy violations, and the fixed files are no longer excluded from our mypy validation.

### CODE REVIEW

**Comment by MichaelSnowden** [comment]
Looks like there are a few spots missing code coverage that I touched. I'll add it

**Comment by MichaelSnowden** [comment]
> Looks like there are a few spots missing code coverage that I touched. I'll add it

This was harder than I thought. I'll just do these 2 files in this PR and tackle the others separately

File: policyengine_core/data/dataset.py [hunk 1]
```diff
@@ -163,8 +166,11 @@ def save(self, year: int, key: str, values: Union[np.array, pd.DataFrame]):
                 f"Invalid data format {self.data_format} for dataset {self.label}."
             )
 
+    # This type variable represents any type that satisfies the Sequence interface.
+    TimeSeriesValues = TypeVar("TimeSeriesValues", bound=Sequence)
```
Comment (nikhilwoodruff) [inline policyengine_core/data/dataset.py:170]: Is this a custom-named type? By the way, the array actually isn't a time series, it's more of the structure:
{
  employment_income: {
    2022: [household_1_value, household_2_value, ...]
  }
}

**Review by nikhilwoodruff — APPROVED** [review]
Thanks @MichaelSnowden for this! Just a minor question, good to merge after resolving.

File: policyengine_core/data/dataset.py (hunk 1 above)
Comment (MichaelSnowden) [reply policyengine_core/data/dataset.py]: I removed this typevar. I think we can just use `Sequence`. Let me know if this looks correct now.

Comment (nikhilwoodruff) [reply policyengine_core/data/dataset.py]: Thanks, LGTM!

**Pull request merged**